spark/v3.1/spark/src/main/java/org/apache/spark/sql/execution.datasources/SparkExpressionConverter.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources
+
+import org.apache.iceberg.spark.SparkFilters
+import org.apache.spark.sql.catalyst.expressions.Expression
+
+object SparkExpressionConverter {
+
+  def convertToIcebergExpression(sparkExpression: Expression): org.apache.iceberg.expressions.Expression = {
+    // Currently, it is a double conversion as we are converting Spark expression to Spark filter
+    // and then converting Spark filter to Iceberg expression.
+    // But these two conversions already exist and well tested. So, we are going with this approach.
+    SparkFilters.convert(DataSourceStrategy.translateFilter(sparkExpression, supportNestedPredicatePushdown = true).get)
+  }
+}
```

Comment:
I noticed that this Scala file was placed in the `java` codepath (e.g. it's in `src/main/java` and not `src/main/scala`).

Is there a reason for this? Seems like something that might need to be updated in the original PR as well.

Reply:
good catch @kbendick , I think I didn't notice it while creating a package. I will first fix in the spark-3.2 and then update these PR. Thanks

Reply:
No worries. I feel like we already have this class (though maybe I saw it when you previously merged it). I believe I saw it being used in the `add_files` procedure code.

If we already do, then just use that one and just remove the extra file.

Thanks for taking care of this. 🙂

Reply:
I checked `add_files` procedure, it doesn't involve functionality or code. So, probably you might have seen it in my spark-3.2 merged PR.